frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx

```diff
@@ -39,36 +45,34 @@ export const meta: MetaFunction<typeof loader> = mergeMeta(({ data }) => {
   return data ? getTitleMetaTags(data.meta.title) : [];
 });
 
-export async function loader({ context: { session }, request }: LoaderFunctionArgs) {
+export async function loader({ context: { session }, request, params }: LoaderFunctionArgs) {
   featureEnabled('edit-personal-info');
+
   const instrumentationService = getInstrumentationService();
   const raoidcService = await getRaoidcService();
-  const userService = getUserService();
 
   await raoidcService.handleSessionValidation(request, session);
 
-  const userId = await userService.getUserId();
-  const userInfo = await userService.getUserInfo(userId);
-
-  if (!userInfo) {
-    instrumentationService.countHttpStatus('phone-number.edit', 404);
-    throw new Response(null, { status: 404 });
-  }
-
   const csrfToken = String(session.get('csrfToken'));
+  const personalInformationRouteHelper = getPersonalInformationRouteHelpers();
 
+  const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  const personalInformation = await personalInformationRouteHelper.getPersonalInformation(userInfoToken, params, request, session);
+
+  session.set('personalInformation', personalInformation);
```

Comment:
I wonder if we should be setting the `personalInformation` in the session here. At this point, `personalInformation` should already be in the session from a previous request in my opinion. 

Maybe we should be setting the `personalInformation` in the session in the `/personal-information` index page instead. Then, if `personalInformation` cannot be found on this edit page, we redirect to `/personal-information`.

Reply:
As discussed, the helper function already redirects the user to `/data-unavailable` if `personalInformation` does not exist in the session or service. 

We can move the `session.set('personalInformation', personalInformation) line in the route helper if `personalInformation` is found from the service.